docs/src/pages/components/tooltips/AnchorElTooltips.tsx

```diff
@@ -27,13 +27,11 @@ export default function AnchorElTooltips() {
       PopperProps={{
         popperRef,
         anchorEl: {
-          clientHeight: 0,
-          clientWidth: 0,
           getBoundingClientRect: () => ({
-            top: areaRef.current?.getBoundingClientRect().top ?? 0,
+            top: areaRef.current!.getBoundingClientRect().top ?? 0,
```

Comment:
If we assume that it'll be defined we no longer need nullish-coalescing: `getBoundingClientRect().top` always returns a number.

Aside: Why don't we just spread it i.e.g `() => ({ ...areaRef.current.getBoundingClientRect(), left: positionRef.current.x, width: 0, ...etc })`? Calling `getBoundingClientRect` twice "looks slow".

Reply:
We can do ...getBoundingClientReact, DOMRect attributes are not iterable.

Reply:
Right, I think properties in DOM aren't enumerable by default.

Interesting alternative (though no IE 11) that creates an actual DOMRect for type safety (and ensures calculations are correct by eliminating duplication): 

```js
getBoundingClientRect = () => DOMReact.fromRect({ x: positionRef.current.x, y: areaRef.current.getBoundingClientRect().y, width: 0, height: 0 })